umbral/keys.py

```diff
@@ -10,13 +10,78 @@
 from cryptography.hazmat.backends.openssl.ec import (
     _EllipticCurvePublicKey, _EllipticCurvePrivateKey
 )
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+
 
 from umbral.config import default_params
-from umbral.point import Point, BigNum
+from umbral.point import Point
+from umbral.bignum import BigNum, hash_to_bn
 from umbral.params import UmbralParameters
 
+from abc import ABC, abstractmethod
+ 
+class AbstractUmbralKeyingMaterial(ABC):
```

Comment:
I don't think an abstract base class is needed here. `UmbralKeyingMaterial` can be made as its own independent class. I think it's far more readable that way.

Reply:
Sounds good.